.github/workflows/win-cpu-x64-build.yml

```diff
@@ -16,13 +16,19 @@ env:
   ort_url: "https://github.com/microsoft/onnxruntime/releases/download/v1.17.0/$(ort_zip)"
 
 jobs:
```

Comment:
Please put a useful description in your pull request, state why this change is needed.

Reply:
This PR is to reduce all future CI pipeline runtime by enabling parallelization on job/steps. This also enable Faile fast by defualt--If one of the step failures, such as build, it will fail all other steps are running in parallel.

Reply:
You mean if one of the job in the "matrix" fails,  the other one will be cancelled immediately ?

Reply:
Is it because initializing CodeQL took about 6 minutes? I think we need to pay the cost. In general CodeQL  should run in the same job, because it needs to gather build information. And it should be added to every our build pipeline. Though you may list CodeQL separately by also duplicating the build part, we will need 2x machines for doing so. Compared to the additional resources we need, the saving isn't too much.